cocos/tween/tween.ts

```diff
@@ -359,10 +367,13 @@ export class Tween<T> {
      * @zh
      * 添加一个隐藏 action，只适用于 target 是节点类型的。
      */
-    hide (): Tween<T> {
-        const action = hide();
-        this._actions.push(action);
-        return this;
+    hide (): TweenWithNodeTargetOrUnknown<T> {
```

Comment:
hide will return `unknown` type if the target is not Node type. And ts compiler will also trigger an error like: 

```
Argument of type 'unknown' is not assignable to parameter of type 'Tween<MyCustomData>'.ts(2345)
```